polymath/polynomial.py

```
@@ -487,26 +608,83 @@ def eval(self, x, recursive=True):
+                # Explicitly preserve the mask from self
+                return Scalar(const_values, mask=self._mask, derivs=derivs, example=self)
             else:
-                return Scalar(example=self.wod)
+                # Use example=self.wod to copy properties without derivatives
+                # Explicitly preserve the mask from self
+                return Scalar(const_values, mask=self._mask, derivs={}, example=self.wod)
 
         x = Scalar.as_scalar(x, recursive=recursive)
         x_powers = [1., x]
         x_power = x
         for k in range(1, self.order):
```

Ruff (0.14.8)

Loop control variable `k` not used within loop body

Rename unused `k` to `_k`

(B007)